src/Randomizer.SMZ3.Tracking/AutoTracking/AutoTracker.cs

```diff
@@ -0,0 +1,695 @@
+            {
+                // CheckZeldaItemMemory
+                CheckLocations(action.CurrentData, LocationMemoryType.ZeldaMisc, false, Game.Zelda);
+            }
+        }
+
+        /// <summary>
+        /// Checks to see if the player has cleared locations in Super Metroid
+        /// </summary>
+        /// <param name="action"></param>
+        protected void CheckMetroidLocations(EmulatorAction action)
+        {
+            if (action.CurrentData != null && !action.HasDataChanged())
+            {
+                CheckLocations(action.CurrentData, LocationMemoryType.Default, false, Game.SM);
+            }
+        }
+
+        /// <summary>
+        ///  Checks if the player has defeated Super Metroid bosses
+        /// </summary>
+        /// <param name="action"></param>
+        protected void CheckMetroidBosses(EmulatorAction action)
+        {
+            if (action.CurrentData != null && !action.HasDataChanged())
+            {
+                CheckSMBosses(action.CurrentData);
+            }
+        }
+
+        /// <summary>
+        /// Checks locations to see if they have accessed or not
+        /// </summary>
+        /// <param name="data">The memory returned from the emulator</param>
+        /// <param name="type">The type of location to find the correct LocationInfo objects</param>
+        /// <param name="is16Bit">Set to true if this is a 16 bit value or false for 8 bit</param>
+        /// <param name="game">The game that is being checked</param>
+        protected void CheckLocations(EmulatorMemoryData data, LocationMemoryType type, bool is16Bit, Game game)
+        {
+            foreach (var location in Tracker.World.Locations.Where(x => x.MemoryType == type && ((game == Game.SM && x.Id < 256) || (game == Game.Zelda && x.Id >= 256))))
+            {
+                try
+                {
+                    var loc = location.MemoryAddress ?? 0;
+                    var flag = location.MemoryFlag ?? 0;
+                    if (!location.Cleared && ((is16Bit && data.CheckUInt16(loc * 2, flag)) || (!is16Bit && data.CheckBinary8Bit(loc, flag))))
+                    {
+                        var item = Tracker.Items.SingleOrDefault(x => x.InternalItemType == location.Item.Type);
+                        if (item != null)
+                        {
+                            Tracker.TrackItem(item, location, null, null, true);
+                            _logger.LogInformation($"Auto tracked {location.Item.Name} from {location.Name}");
+                        }
+                        else
+                        {
+                            Tracker.Clear(location, null, true);
+                            _logger.LogInformation($"Auto tracked {location.Name} as cleared");
+                        }
+                    }
+
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError("Unable to auto track location: " + location.Name);
```

Comment:
Use the overload that starts with the exception. It should print the message and stacktrace automatically.